src/main/java/kr/syeyoung/dungeonsguide/features/impl/party/playerpreview/FeatureViewPlayerStatsOnJoin.java

```diff
@@ -0,0 +1,631 @@
+            playerProfile = Optional.ofNullable(currentrySelectedProfile);
+        }
+
+
+        draw(scaledResolution, mouseX, mouseY, plsSetAPIKEY, playerProfile);
+    }
+
+    private void draw(ScaledResolution scaledResolution, int mouseX, int mouseY, boolean plsSetAPIKEY, Optional<PlayerProfile> playerProfile) {
+        GlStateManager.pushMatrix();
+        GlStateManager.translate(popupRect.x, popupRect.y, 0);
+        Gui.drawRect(0, 0, popupRect.width, popupRect.height, 0xFF23272a);
+        Gui.drawRect(2, 2, popupRect.width - 2, popupRect.height - 2, 0XFF2c2f33);
+
+        GlStateManager.enableBlend();
+        GL14.glBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_ALPHA);
+        GlStateManager.tryBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_ALPHA);
+        if (plsSetAPIKEY) {
+            mc.fontRendererObj.drawString("Please set API KEY on /dg -> Party Kicker", 5, 5, 0xFFFFFFFF);
+            GlStateManager.popMatrix();
+        } else if (playerProfile == null) {
+            mc.fontRendererObj.drawString("Fetching data...", 5, 5, 0xFFFFFFFF);
+            GlStateManager.popMatrix();
+        } else if (!playerProfile.isPresent()) {
+            mc.fontRendererObj.drawString("User could not be found", 5, 5, 0xFFFFFFFF);
+            GlStateManager.popMatrix();
+        } else {
+            int relX = mouseX - popupRect.x;
+            int relY = mouseY - popupRect.y;
+            FontRenderer fr = mc.fontRendererObj;
+            GL11.glEnable(GL11.GL_SCISSOR_TEST);
+            clip(scaledResolution, popupRect.x, popupRect.y, popupRect.width, popupRect.height);
+            Gui.drawRect(0, 168, 90, 195, 0xFF23272a);
+            Gui.drawRect(2, 170, 88, 193, new Rectangle(2, 170, 86, 23).contains(relX, relY) ? 0xFFff7777 : 0xFFFF3333);
+            Gui.drawRect(0, 193, 90, 220, 0xFF23272a);
+            Gui.drawRect(2, 195, 88, 218, new Rectangle(2, 195, 86, 23).contains(relX, relY) ? 0xFF859DF0 : 0xFF7289da);
+            Gui.drawRect(90, 193, 180, 220, 0xFF23272a);
+            Gui.drawRect(91, 195, 178, 218, new Rectangle(91, 195, 176, 23).contains(relX, relY) ? 0xFFdace72 : 0xdddace72);
+            GlStateManager.enableBlend();
+            GL14.glBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_ALPHA);
+            GlStateManager.tryBlendFuncSeparate(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, GL11.GL_ONE, GL11.GL_ONE_MINUS_SRC_ALPHA);
+            fr.drawString("Kick", (90 - fr.getStringWidth("Kick")) / 2, (364 - fr.FONT_HEIGHT) / 2, 0xFFFFFFFF);
+            fr.drawString("Invite", (90 - fr.getStringWidth("Invite")) / 2, (414 - fr.FONT_HEIGHT) / 2, 0xFFFFFFFF);
+            fr.drawString("Change Profile", ((90 - fr.getStringWidth("Change Profile")) / 2) + 90, (414 - fr.FONT_HEIGHT) / 2, 0xFFFFFFFF);
```

Comment:
A room for improvement
- Hovering highlights button
- maybe put the button next to the view inventory button? or arrows on the side of player